### PR TITLE
(#4086) - removeLocal() during bulkDocs()

### PR DESCRIPTION
**Pull request by nolanlawson**
This increases our code coverage very slightly
by allowing the user to remove a `_local` doc
during a `bulkDocs()`.

Incidentally this test led me to find a bug in
CouchDB 1.6.1. I'll report it in their bugtracker.

### CODE REVIEW

**Comment by nolanlawson** [comment]
I rebased this on #4082 to avoid bitrotting myself.

**Comment by nolanlawson** [comment]
I'll note that since removing a local doc inside of bulkDocs is something that our users might never do (and even CouchDB doesn't accept it), then another valid solution to the coverage problem would be to remove the `removeLocal` code from `processDocs.js`.

**Comment by daleharvey** [comment]
Does couchdb-master let you delete a local doc in bulkDocs? it sounds like a reasonable thing to do so this is probably good, but would like to make sure we arent breaking compatibility with Couch on purpose here, @rnewson ?

File: tests/integration/test.bulk_docs.js [hunk 1]
```diff
@@ -320,6 +320,89 @@ adapters.forEach(function (adapter) {
...
+        return db.allDocs();
+      }).then(function (info) {
+        info.rows.should.have.length(0);
+        return db.get('_local/godzilla').then(function () {
+          throw new Error('expected 404');
+        }, function (err) {
+          should.exist(err);
+        });
+      });
+    });
```
Comment (nolanlawson) [inline tests/integration/test.bulk_docs.js]: ^ @daleharvey This test passes on CouchDB 1.6, so yes, it lets you delete a _local doc within _bulk_docs.

**Comment by nolanlawson** [comment]
Weird, GitHub is still showing 2 commits instead of only 1, even though this branch is up-to-date. Sorry for the noise.

**Comment by daleharvey** [comment]
Yeh I have had that problem with github before, but yeh if its a plain bug thats failing on couch then :+1: to this, looks great, Did you file the bug? I think if would be very useful to have a reference next to the test so we can enable it when couch gets fixed

**Comment by nolanlawson** [comment]
Yup I did: https://issues.apache.org/jira/browse/COUCHDB-2758

I can add a reference to the bug in the test comments when I merge.

**Comment by rnewson** [comment]
curl foo:bar@localhost:15984/db1/_bulk_docs -Hcontent-type:application/json -d '{"docs":[ {"_id":"_local/foo"}   ]}'
[{"ok":true,"id":"_local/foo","rev":"0-1"}]
➜  ~  curl foo:bar@localhost:15984/db1/_local/foo
{"_id":"_local/foo","_rev":"0-1"}

**Comment by rnewson** [comment]
curl foo:bar@localhost:15984/db1/_bulk_docs -Hcontent-type:application/json -d '{"docs":[ {"_id":"_local/foo","_deleted":true}   ]}'
[{"ok":true,"id":"_local/foo","rev":"0-0"}]
➜  ~  curl foo:bar@localhost:15984/db1/_local/foo
{"error":"not_found","reason":"missing"}

**Comment by nolanlawson** [comment]
@rnewson That's weird; I do think I found a test case that's failing. Did you look at that jira?

**Comment by daleharvey** [comment]
I added a reference to the CouchDB issue and merged, cheers

https://github.com/pouchdb/pouchdb/commit/43d50f675d8d447eb3ce6fa75c409ce1a0c22216

**Pull request closed without merging**